react/src/components/PostHogFeature.tsx

```diff
@@ -0,0 +1,94 @@
+import { useFeatureFlagPayload, useFeatureFlagVariantKey, usePostHog } from '../hooks'
+import React, { RefObject, useEffect, useMemo, useRef, useState } from 'react'
+import { PostHog } from '../context'
+
+export type PostHogFeatureProps = {
+    flag: string
+    children: React.ReactNode | ((payload: any) => React.ReactNode)
+    match?: string | boolean
+    visibilityObserverOptions?: IntersectionObserverInit
+}
+
+export function PostHogFeature({
+    flag,
+    match,
+    children,
+    visibilityObserverOptions,
+}: PostHogFeatureProps): JSX.Element | null {
+    const posthog = usePostHog()
+    const payload = useFeatureFlagPayload(flag)
+    const variant = useFeatureFlagVariantKey(flag)
+    const [clickTracked, setclickTracked] = useState(false)
+
+    if (match === undefined || variant === match) {
+        const childNode: React.ReactNode = typeof children === 'function' ? children(payload) : children
+        return (
+            <div
+                onClick={() => {
+                    if (!clickTracked) {
+                        trackClicks(flag, posthog)
+                        setclickTracked(true)
+                    }
+                }}
+            >
+                <VisibilityTracker flag={flag} options={visibilityObserverOptions}>
+                    {childNode}
+                </VisibilityTracker>
+            </div>
+        )
+    }
+
+    return null
+}
+
+function trackClicks(flag: string, posthog?: PostHog) {
+    posthog?.capture('$feature_flag_clicked', { feature_flag: flag, $set: { [`$feature_interaction/${flag}`]: true } })
```

Comment:
question: When you say `feature_flag_clicked` here, does this mean the component with the feature flag has been clicked? Or something else? Feature flag clicked here sounds a bit vague

Reply:
request: Don't know if this is too much for now or not possible, but it would be nice if we could incorporate which component based on classname or data-attr or something was clicked/interacted with 😃 as one feature flag implementation could involve multiple `<PostHogFeature />` components

Reply:
Possible, & makes some sense, but will skip for now. You wouldn't lose this data, as that's what autocapture will capture anyway. We need to test if that makes sense for flag autocapture too, or if this is sufficient.